### PR TITLE
Added new shape families for Archimedean, Catalan, Johnson, and other solids.

### PR DESCRIPTION
**Pull request by janbridley**
## Description
Expanded on coxeter.families with methods and data for several families of common convex polyhedra:
* ``ArchimedeanFamily``: The 13 Archimedean solids.
* ``CatalanFamily``: The 13 Archimedean duals.
* ``JohnsonFamily``: The 92 Johnson solids.
* ``PrismAntiprismFamily``: The 16 n-gonal prisms and antiprisms ranging from n=3 to n=10.
* ``PyramidDipyramidFamily``: The 6 equilateral pyramids and dipyramids.

Each new family is queryable with ``get_shape("Shape Name")`` and returns a ``ConvexPolyhedron`` normalized to volume of 1, just as ``PlatonicFamily`` currently does. Vertex data for all new families is stored as a float64, and stored data for the current ``PlatonicFamily`` has been updated to match that precision. This change fixes intermittent bugs that arose from lack of numerical precision. New pytests have been added to verify the new families behave as expected. 

## Motivation and Context
Querying polyhedra other than Platonic solids could be relatively difficult. This release streamlines the process and allows for simple access to a greatly increased number of convex solids.

## Types of Changes
- [ ] Documentation update
- [x] Bug fix
- [x] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:
- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/coxeter/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/coxeter/blob/master/ContributorAgreement.md).
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [x] I have updated the [changelog](https://github.com/glotzerlab/coxeter/blob/master/ChangeLog.txt) and the [credits](https://github.com/glotzerlab/coxeter/blob/master/doc/source/credits.rst).

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -7,10 +7,37 @@
...
 
+# Define a function to combine marks in order to more compactly test shape families
+def combine_marks(*marks):
+    combinedargvalues = []
+    combinedids = []
+    for mark in marks:
+        argvalues, ids = mark.kwargs["argvalues"], mark.kwargs["ids"]
+        combinedargvalues.extend(argvalues)
+        combinedids.extend(ids)
+    print(combinedids)
```
Comment (vyasr) [inline tests/conftest.py:31]: I assume this print can be removed now.

File: tests/conftest.py [hunk 2]
```diff
@@ -7,10 +7,37 @@
...
+    PlatonicFamily,
+    PrismAntiprismFamily,
+    PyramidDipyramidFamily,
+    RegularNGonFamily,
+)
 from coxeter.shapes import ConvexPolyhedron, ConvexSpheropolyhedron, Polyhedron, Shape2D
 
 
+# Define a function to combine marks in order to more compactly test shape families
+def combine_marks(*marks):
```
Comment (vyasr) [inline tests/conftest.py:24]: This is neat! I'm kind of surprised that this doesn't already exist, or that I haven't had to implement it myself before.

**Review by vyasr — CHANGES_REQUESTED** [review]
Some minor suggestions here, as well as a request to clean out miniball-related changes. Aside from that, is there anyone else in the group who could confirm the validity of the shape families themselves? I am not working with these currently and it would be good for someone who is to take a few minutes to spot check that the produced shapes are valid.

File: tests/conftest.py (hunk 2 above)
Comment (janbridley) [reply tests/conftest.py:24]: Thank you! It took a little bit to implement but I will definitely be holding onto this code for future projects.

**Review by vyasr — APPROVED** [review]
This PR LGTM now! For future reference, once you've addressed PR review changes you can re-request review by clicking on the button right next to the reviewer name in the Reviewers section on the top-right of the PR page.

**Comment by vyasr** [comment]
@janbridley feel free to click the "Squash and merge" button when you're ready!

File: tests/conftest.py [hunk 3]
```diff
@@ -198,6 +224,72 @@ def platonic_solids():
...
 
+def archimedean_solids():
+    """Generate archimedean solids."""
+    for shape_name in ArchimedeanFamily.data:
+        yield ArchimedeanFamily.get_shape(shape_name)
+
+
+# A convenient mark decorator that also includes names for the polyhedra.
+# Assumes that the argument name is "poly".
+_archimedean_shape_names = ArchimedeanFamily.data.keys()
```
[the remark this answers is not in the document]
Comment (janbridley) [reply tests/conftest.py:235]: Thanks for the feedback, and noted for next time!

**Pull request merged**